### ISSUE TITLE
Incomplete/buggy example for custom-domain-with-self-managed certs

### ISSUE THREAD

**Issue by pablogamboa** [auth0/terraform-provider-auth0#557]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I'm pretty sure the example provided in the docs [is incomplete](https://github.com/auth0/terraform-provider-auth0/blob/main/examples/custom-domain-with-self-managed-certs/main.tf#L10-L30). As part of the domain verification you need to create a TXT record to verify your resource, that's nowhere present in the example code.

Also I think there's more work to be done in the plumbing. When you are using `self_managed_certs` in `auth0_custom_domain` the `auth0_custom_domain.cointracker_custom_domain.verification[0].methods[0].record` returns the uuid to be used in the TXT verification, not the CNAME target. So looks [like this line](https://github.com/auth0/terraform-provider-auth0/blob/main/examples/custom-domain-with-self-managed-certs/main.tf#L16) is incorrect.

### Expectation

A working example that can be used as it is.

### Reproduction

When I translate the sample code to Cloudflare, the domain never gets verified because the TXT record is missing. Also when I try to create the CNAME pointing to my domain, i.e. this part:

```terraform
resource "google_dns_record_set" "my_domain_verification" {
  name         = "${auth0_custom_domain.my_domain.verification[0].methods[0].domain}."
  managed_zone = var.managed_zone_name
  type         = upper(auth0_custom_domain.my_domain.verification[0].methods[0].name)
  ttl          = 300
  rrdatas = [
    "${auth0_custom_domain.my_domain.verification[0].methods[0].record}.",
  ]
}
```

it fails because it's not a valid DNS value (it's a uuid)

### Auth0 Terraform Provider version

0.43.0

### Terraform version

1.3.7

**Comment by pablogamboa** [comment]
If it helps, I've created a [blog post](https://pablomarti.dev/auth0-custom-domain-with-cloudflare/) with a working example for custom domains + Cloudflare. The current example in the docs can't work, at least a TXT record is missing and a load balancer must be created.

**Comment by sergiught** [comment]
Hey @pablogamboa 👋🏻 

I'm closing this down as we decided to remove those examples from the repo as they weren't getting propagated to the terraform registry guides. We'll look into migrating them as proper guides in the foreseeable future. Contributions are also welcome! 

Also we shipped the [v1.0.0-beta.0](https://registry.terraform.io/providers/auth0/auth0/latest/docs) release! 🙌🏻 Please give it a try and let us know if you encounter any issues.

```
terraform {
  required_providers {
    auth0 = {
      source  = "auth0/auth0"
      version = "1.0.0-beta.0"
    }
  }
}
```

### 📚 Resources

[Quickstart](https://registry.terraform.io/providers/auth0/auth0/latest/docs/guides/quickstart) | [Documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) | [Migration Guide](https://github.com/auth0/terraform-provider-auth0/blob/v1/MIGRATION_GUIDE.md) | [Changelog](https://github.com/auth0/terraform-provider-auth0/releases/tag/v1.0.0-beta.0)
